### ISSUE TITLE
Artifacts occur in some pictures

### ISSUE THREAD

**Issue by LAccordeur** [thstkdgus35/EDSR-PyTorch#57]
Hi, I tried your code these days and most results were awesome. Thank you for your work!
I use the same method of downsample for these pictures, however, there are some artifacts in a few pictures, especially in pictures such as buildings or regular texture. Here is the example. Could you help me?

- origin
![img_010_srf_2_hr](https://user-images.githubusercontent.com/16172480/44402917-ac9cf100-a585-11e8-91b3-7e2471dea3b9.png)

- edsr
![img_010_edsr](https://user-images.githubusercontent.com/16172480/44402935-b9214980-a585-11e8-9583-8afb38fddd49.png)

- origin
![img_003_srf_2_hr](https://user-images.githubusercontent.com/16172480/44403035-f259b980-a585-11e8-838b-2447f6da6574.png)

- edsr
![img_003_edsr](https://user-images.githubusercontent.com/16172480/44402993-db1acc00-a585-11e8-9d63-7e15a8c0a759.png)

**Comment by thstkdgus35** [comment]
Hello.

Would you upload the original high-resolution images with input low-resolution images?

Thank you.

**Comment by LAccordeur** [comment]
The model I used is EDSR baseline(x2). I find that the picture would has this problem when I used the following command to down sample.
` img = cv2.resize(img, None, fx=0.5, fy=0.5, interpolation=cv2.INTER_CUBIC`

**Comment by thstkdgus35** [comment]
Have you tried the MATLAB bicubic function?

It is known that MATLAB has a different bicubic implementation compared to the other libraries.

Although I checked that our models (trained with MATLAB bicubic) work well with scipy bicubic function, it would be better to try MATLAB bicubic function first to find out the problem.

**Comment by LAccordeur** [comment]
Yes, I had tried the MATLAB bicubic function, which didn't have this problem. It's actually caused by the different implementations of bicubic. Thank you for your help.

**Comment by LAccordeur** [comment]
I am confused about why you choose the MATLAB bicubic function to generate train set. Is it has some advantages compared to other functions such as cv2.resize()?

**Comment by thstkdgus35** [comment]
It is just because MATLAB bicubic is a standard for many previous super-resolution papers.

By the way, it seems weird because I checked that our models work well with other bicubic implementation (scipy imresize).

**Comment by LAccordeur** [comment]
Yes, your models work well with scipy imresize. I also tried it before. 

Have you tried opencv's bicubic implementation?  As I mentioned before, I encountered this problem when I used the following command. I supposed that opencv's implementation is different from the implementation above. 
`img = cv2.resize(img, None, fx=0.5, fy=0.5, interpolation=cv2.INTER_CUBIC`

**Comment by thstkdgus35** [comment]
OK. I will check that issue.

**Comment by Ir1d** [comment]
@LAccordeur Hi, have you noticed the reason of this?
May I ask if using [torchvision.transforms.Resize](https://pytorch.org/docs/0.4.0/torchvision/transforms.html?highlight=resize#torchvision.transforms.Resize) results the same?
Thx in advance.

**Comment by LAccordeur** [comment]
@Ir1d Hi, I'm sorry that I haven't used torchvision.transforms.Resize before, so I don't know if this implement would result the same. Perhaps the different implements of different libraries cause this problem, and here is another  #34 issue about  this problem and maybe you can find something.
